src/DssSpell.sol

```diff
@@ -65,312 +70,293 @@ contract DssSpellAction is DssAction {
+        // Forum: https://forum.sky.money/t/atlas-core-development-payment-requests-november-2024/25580
+        // Atlas: https://forum.sky.money/t/atlas-core-development-payment-requests-november-2024/25580/8
 
-        // Bonapublica - 2.06 MKR - 0x167c1a762B08D7e78dbF8f24e5C3f1Ab415021D3
-        MKR.transfer(BONAPUBLICA, 2.06 ether); // Note: 'ether' is a keyword helper, only MKR is transferred here
+        // Kohla (Cloaky) - 20,000 USDS - 0x73dFC091Ad77c03F2809204fCF03C0b9dccf8c7a
+        _transferUsds(CLOAKY_KOHLA_2,  20_000 * WAD);
 
-        // Rocky - 1.17 MKR - 0xC31637BDA32a0811E39456A59022D2C386cb2C85
-        MKR.transfer(ROCKY, 1.17 ether); // Note: 'ether' is a keyword helper, only MKR is transferred here
+        // Ennoia (Cloaky) - 20,110 USDS - 0xA7364a1738D0bB7D1911318Ca3FB3779A8A58D7b
+        _transferUsds(CLOAKY_ENNOIA,   20_110 * WAD);
+
+        // BLUE - 50,167 USDS - 0xb6C09680D822F162449cdFB8248a7D3FC26Ec9Bf
+        _transferUsds(BLUE,            50_167 * WAD);
+
+        // BLUE - 330,000 SKY - 0xb6C09680D822F162449cdFB8248a7D3FC26Ec9Bf
+        _transferSky( BLUE,           330_000 * WAD);
 
         // ---------- Spark Proxy Spell ----------
-        // Forum: https://forum.sky.money/t/14-nov-2024-proposed-changes-to-spark-for-upcoming-spell/25466
-        // Poll: https://vote.makerdao.com/polling/QmQizL1F
-        // Poll: https://vote.makerdao.com/polling/Qmbohkr5
-        // Poll: https://vote.makerdao.com/polling/QmYqM8Yf
-        // Poll: https://vote.makerdao.com/polling/QmXsXzot
-        // Poll: https://vote.makerdao.com/polling/Qmf955yA
-
-        // Execute Spark Proxy Spell at 0x8a3aaeAC45Cf3D76Cf82b0e4C63cCfa8c72BDCa7
+        // Forum: https://forum.sky.money/t/stability-scope-parameter-changes-18-sfs-dsr-ssr-spark-effective-dai-borrow-rate-spark-liquidity-layer/25593
+        // Forum: https://forum.sky.money/t/28-nov-2024-proposed-changes-to-spark-for-upcoming-spell/25543
+        // Poll: https://vote.makerdao.com/polling/QmSxJJ6Z
+        // Poll: https://vote.makerdao.com/polling/QmaxFZfF
+        // Poll: https://vote.makerdao.com/polling/QmWUkstV
+        // Poll: https://vote.makerdao.com/polling/QmQ2Umfm
+        // Poll: https://vote.makerdao.com/polling/QmcNd4mH
+
+        // Note: The spark spell below expects the new auto-line settings to be effective.
+        DssAutoLineAbstract(AUTO_LINE).exec("ALLOCATOR-SPARK-A");
+
+        // Execute Spark Proxy Spell at 0x6c87D984689CeD0bB367A58722aC74013F82267d
         ProxyLike(SPARK_PROXY).exec(SPARK_SPELL, abi.encodeWithSignature("execute()"));
     }
+
+    /// @notice wraps the operations required to transfer USDS from the surplus buffer.
+    /// @param usr The USDS receiver.
+    /// @param wad The USDS amount in wad precision (10 ** 18).
+    function _transferUsds(address usr, uint256 wad) internal {
+        // Note: Enforce whole units to avoid rounding errors
+        require(wad % WAD == 0, "transferUsds/non-integer-wad");
```

Comment:
If we're enforcing whole units, why not take whole units as an input like `sendPaymentFromSurplusBuffer` does?

Reply:
Those functions are likely candidates for inclusion into the DssExecLib, so I would also name it appropriately, e.g.: `sendUsdsPaymentFromSurplusBuffer`

Reply:
I dont untherstand , for example